components/list/list-item-generic-layout.js

```diff
@@ -450,7 +450,8 @@ class ListItemGenericLayout extends RtlMixin(LitElement) {
 				let nestedList;
 				do {
 					// this check needs to account for standard list-items as well as custom
-					nestedList = previousElement.querySelector('[slot="nested"]') || previousElement.shadowRoot.querySelector('d2l-list');
+					nestedList = previousElement.shadowRoot.querySelector('d2l-list');
+
 					if (nestedList) {
 						const nestedListItems = [...nestedList.children].filter(node => node.role === 'rowgroup');
 						if (nestedListItems.length) {
```

Comment:
Ok, it looks like we're assuming that if there's a nested list, it will have items so that `previousElement` is reassigned for a new query. This code is trying to drill into the deepest last item since it would be the previous visual item (flattening). If there is a nested list with no items, it will skip this assignment and never exit the loop.

I think we could make this code a bit smarter with an `else` condition here. But first, just to confirm, are you rendering lists with no items? If so, do you need to/are you sure you want to?

Reply:
This is a good question/call out^. I took a deeper look into it and we do end up rendering a nested `d2l-list` with no items in it for our bottom tree leaf.

I don't think that we need to be rendering a nested `d2l-list` with no items in it.
@MitchelPaulin @semibran Any concerns with this approach? 

Here is a PR with these changes in our `d2l-outcomes` repo.
https://github.com/Brightspace/d2l-outcomes/pull/1447

If Brandon/Mitch have no concerns then I think it makes more sense to handle this in our code. 
I'm open to different approaches here as well. Let me know.

Reply:
Yeah, I mean if that's not necessary then I'd probably expect the app to not render the empty lists, if for no other reason than limiting the amount of stuff being rendered - even an empty `d2l-list` would have some rendering cost since it has some initialization logic. I'm also not totally sure how those empty lists would announce using a screen reader - that may affect your decision.

Having said that, I will create a Gaudi ticket to make this code a bit more resilient. If you end up depending on that fix, let me know so we can prioritize.

Reply:
Gaudi ticket: [GAUD-6690](https://desire2learn.atlassian.net/browse/GAUD-6690)

Reply:
Yeah this makes sense to me. Let's go with the approach I have in https://github.com/Brightspace/d2l-outcomes/pull/1447. 

Thanks for filing a Jira ticket here. I don't think we need this ticket prioritized at the moment since I have a workaround. I'm going to close this PR off for now. Feel free to use this PR/the previous outcomes PR as an example for fixing the Jira ticket you logged.

Reply:
Awesome. Thanks for digging into this Jesse. It was a huge help. 😄